##mapping trimmed reads to vOTU

#dir of trimmed reads

#mapping to glaicer viromes
```
#wkdir
cd /home/liupf/Projects/glacier_virome/mapping_to_MAGS1020
```


#reads metafile
```
/home/liupf/Projects/glacier_virome/mapping_glacier_vOTU/

cat mapping_metafile_TP85.txt mapping_metafile_partII_III.txt mapping_metafile_non-TP45.txt >mapping_metafile_TP112_nonTP45.txt




```

##prepare MAGs
```
#1020 MAGs; TP MAGs
#D1.0.1_host_preparation.md

cd /home/liupf/Projects/glacier_virome/MAGs-TP-2661ana

for file in $(cat ./MAGs_1020_drep099.txt)
do
ln -s /home/ptpe/project_raw_data/glacier_virome_rawdata/2661_TGG_MAGs/${file} /home/liupf/Projects/glacier_virome/mapping_to_MAGS1020
done
```

#non-TP 344 MAGs

```
cd /home/liupf/Projects/glacier_virome/MAGs-Non-TP-405ana

#get the 0.99 cluster files a
nano NonTP-MAGs_344_drep099.txt

for file in $(cat NonTP-MAGs_344_drep099.txt)
do
ln -s /home/ptpe/project_raw_data/glacier_virome_rawdata/Non-TP-glacierMAGs405/${file} /home/liupf/Projects/glacier_virome/mapping_to_MAGS1020
done


#reformate seqs head
for file in *.fa
do

sed -e "s/>/>"${file%.*}_Gx"/g" ${file} > "${file%.*}".fna

sed -i -e 's/ .*//g' "${file%.*}".fna
done

#merge all genome and do mapping
cat *.fna > ref.fasta
```

### mapping
```
##do mapping
cd /home/liupf/Projects/glacier_virome/mapping_to_MAGS1020

screen -r bbmaping_to_MAGs


#bbmap_glacier_MAGs.sh <metafile> <threads> <ref>
#./bbmap_glacier_MAGs.sh /home/liupf/Projects/glacier_virome/mapping_glacier_vOTU/mapping_metafile_TP112_nonTP45.txt 40 ${file}

./bbmap_glacier_MAGs_merge.sh /home/liupf/Projects/glacier_virome/mapping_glacier_vOTU/mapping_metafile_TP112_nonTP45.txt 50 ref.fasta

```

### 20210803
```
#mapping to MAGs with virus from crispr prediciton

cd /home/liupf/Projects/glacier_virome/mapping_to_MAGS1020

nano MAGs_w_virus.txt

for file in $(cat MAGs_w_virus.txt)
do echo ${file}
cat ${file} >> ref_w_virus.fasta
done
#28426 seqs; matched

screen -S bbmap2MAGs

#restart at 20210813
./bbmap_glacier_MAGs_merge.sh /home/liupf/Projects/glacier_virome/mapping_glacier_vOTU/mapping_metafile_TP112_nonTP45.txt 50 ref_w_virus.fasta &>mapping_log2.txt


# bam to sorted bam
mkdir MAGs78_w_virus_bam
cd /home/liupf/Projects/glacier_virome/mapping_to_MAGS1020/MAGs78_w_virus_bam

mv ../*.bam ./

conda activate coverm
for bamfile in *.bam
do
samtools sort -@ 50 "${bamfile}" > "${bamfile}".sorted
done

rm *.bam

#mean coverage
coverm genome --bam-files *bam.sorted --genome-fasta-directory  ../ -x fna -o MAGs78_Sample166_mean_output-merge.tsv -m mean -t 50

#tpm
coverm genome --bam-files *bam.sorted --genome-fasta-directory  ../ -x fna -o MAGs78_Sample166_TPM_output-merge.tsv -m tpm -t 50

coverm genome --bam-files *bam.sorted --min-covered-fraction 0 --genome-fasta-directory  ../ -x fna -o MAGs78_Sample166_count_output-merge.tsv -m count -t 15

```


##20210818
```
#mapping to MAGs that predected as host by tRNA blastn and VirHostMatcher

# list of host MAGs
# /Users/pengfeiliu/A_TBP/文章-MS-TP/glacier-virome2021/virus-host-prediction/glacier_vhm_best_host_1364MAGs.xlsx/host_tRNA_vhm

#
cd /home/liupf/Projects/glacier_virome/mapping_to_MAGS1020

mkdir MAGs1042_w_virus_bam

nano MAGs1042_w_virus.txt

for file in $(cat MAGs1042_w_virus.txt)
do echo ${file}
cat ${file} >> ref1042_w_virus.fasta
done

#grep -c '>' ref1042_w_virus.fasta
#366256

screen -S bbmap2MAGs

#restart at 20210831
./bbmap_glacier_MAGs_merge.sh /home/liupf/Projects/glacier_virome/mapping_glacier_vOTU/mapping_metafile_TP112_nonTP45.txt 50 ref1042_w_virus.fasta &>ref1042_w_virus_6.txt

#-rw-rw-r-- 1 ptpe ptpe  25G Aug 31 18:21 M-PL4snow.bam

mkdir mapping_second_round

conda activate coverm
for bamfile in *.bam
do
samtools sort -@ 50 "${bamfile}" > "${bamfile}".sorted
done

mv *sorted mapping_second_round
```

#
```
#mean coverage
coverm genome --bam-files *bam.sorted --genome-fasta-directory  ../ -x fna -o MAGs1042_Sample166_mean_output-merge.tsv -m mean -t 50

#tpm
coverm genome --bam-files *bam.sorted --genome-fasta-directory  ../ -x fna -o MAGs1042_Sample166_TPM_output-merge.tsv -m tpm -t 50

##
#
coverm genome --bam-files *bam.sorted --min-covered-fraction 0 --genome-fasta-directory  ../ -x fna -o MAGs1042_Sample166_count_output-merge.tsv -m count -t 20

```


##mapping to the rest of genomes
#
```
cd /home/liupf/Projects/glacier_virome/mapping_to_MAGS1020

nano MAGs_v_virus1120.list
grep -v -w -f MAGs_v_virus1120.list  MAGs_1364.list > non-host-MAGs.txt
#244

for file in $(cat non-host-MAGs.txt)
do #echo ${file}
#cat "${file}".fna >> ref244_non_host.fasta
grep -c '>' "${file}".fna
done

#grep -c '>' ref244_non_host.fasta
#91490

screen -r bbmap2MAGs

#restart at 20210910
./bbmap_glacier_MAGs_merge.sh /home/liupf/Projects/glacier_virome/mapping_glacier_vOTU/mapping_metafile_TP112_nonTP45.txt 50 ref244_non_host.fasta &>ref244_w_virus_2.txt

mkdir mapping_third_round
mv *sorted mapping_third_round

#
## mean coverage
coverm genome --bam-files *bam.sorted --genome-fasta-directory  ../ -x fna -o MAGs244_Sample166_mean_output-merge.tsv -m mean -t 50

## tpm
coverm genome --bam-files *bam.sorted --genome-fasta-directory  ../ -x fna -o MAGs244_Sample166_TPM_output-merge.tsv -m tpm -t 50

#
coverm genome --bam-files *bam.sorted --min-covered-fraction 0 --genome-fasta-directory  ../ -x fna -o MAGs244_Sample166_count_output-merge.tsv -m count -t 20
```

##cat all sorted bam together
````
for file in $(cat sorted_bam.list)
do
echo "${file}"
samtools merge -@ 30 -f ${file} ./mapping_second_round/$file ./mapping_third_round/$file
#cat ./MAGs78_w_virus_bam/$file ./mapping_second_round/$file ./mapping_third_round/$file  >${file}
done

#not working well
#some errors 20210917



#
## mean coverage
coverm genome --bam-files *bam.sorted --genome-fasta-directory  ./ -x fna -o MAGs_Sample166_mean_output-merge.tsv -m mean -t 50

## tpm
coverm genome --bam-files *bam.sorted --genome-fasta-directory  ./ -x fna -o MAGs_Sample166_TPM_output-merge.tsv -m tpm -t 50

#count

```

##20210922
```
#from counts to TPM
#on MAC

cd /Users/pengfeiliu/A_TBP/文章-MS-TP/glacier-virome2021/MAGs/relative_abundances_MAGs/

grep -w -f MAGs1042_w_virus.txt MAGs1042_Sample166_count_output-merge.tsv > MAGs1042_Sample166_count_output-short.tsv
#1045
ISO_RC12-3-1.002
ISO_DWP2-2-2

grep -w -f MAGs244_wo_virus.txt MAGs244_Sample166_count_output-merge.tsv > MAGs244_Sample166_count_output-short.tsv
#245 MAGs244_Sample166_count_output-short.tsv

grep -w -f MAGs78_w_virus.txt MAGs78_Sample166_count_output-merge.tsv > MAGs78_Sample166_count_output-short.tsv
#79

cat MAGs78_Sample166_count_output-short.tsv MAGs1042_Sample166_count_output-short.tsv MAGs244_Sample166_count_output-short.tsv > MAGs_Sample166_count_output-short.tsv
```



##coverM test
```
for bamfile in *.bam
do
samtools sort -@ 10 "${bamfile}" > "${bamfile}".sorted
done

coverm genome --bam-files *bam.sorted --genome-fasta-directory  ../ -x fna -o mean_output-merge.tsv -m mean


coverm genome --bam-files *bam.sorted -r ../ERR1201173_maxbin2_bin.0.fna -x fna -o mean_output2.tsv -m mean

```

###
##mapping to the rest of MAGs (w/o cripspr host prediction)
##these may other means of virus-host prediction (tRNA, virhostmatcher, blastn)
